src/app/components/core/system-alert/system-alert.component.ts

```diff
@@ -0,0 +1,128 @@
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-system-alert',
+  templateUrl: './system-alert.component.html',
+  styleUrls: ['./system-alert.component.css']
+})
+
+export class SystemAlertComponent {
+  @Input() alertCount: number;
+  systemAlerts = SystemAlerts['alerts'];
+  public reenableButton = new EventEmitter<boolean>(false);
+
+  constructor(
+    private alertService: AlertService,
+    private router: Router,
+    private systemAlertService: SystemAlertService,
+    public ngProgress: ProgressBarService,
+    private rolesService: RolesService) {
+  }
+
+  ngOnInit() {}
+
+  public toggleDropdown() {
+    const dropDown = document.querySelector('#system-alert-dd');
+    dropDown.classList.toggle('is-active');
+    if (dropDown.classList.contains('is-active')) {
+      this.getAlerts();
+    }
+  }
+
+  getAlerts() {
+    this.systemAlertService.getAlerts().
+    subscribe(
+      (data) => {
+        this.systemAlerts = data['alerts'];
+      },
+      error => {
+        if (error.status === 0) {
+          console.log('service down ', error);
+        } else {
+          this.alertService.error(error.statusText);
+        }
+      }
+    );
+  }
+
+  performAction(alert: SystemAlert) {
+    if (this.getButtonText(alert.message) === 'Show Logs') {
+      this.router.navigate(['logs/syslog'], { queryParams: { source: alert.key } });
+    }
+    if (this.getButtonText(alert.message) === 'Upgrade') {
+      this.update();
+    }
+    this.toggleDropdown();
+  }
+
+  update() {
+    this.ngProgress.start();
+    this.systemAlertService.update()
+    .subscribe(
+      (data) => {
+        this.ngProgress.done();
+        this.reenableButton.emit(false);
+        this.alertService.success(data['status']);
+      },
+      error => {
+        this.ngProgress.done();
+        this.reenableButton.emit(false);
+        if (error.status === 0) {
+          console.log('service down', error);
+        } else {
+          this.alertService.error(error.statusText);
+        }
+      });
+  }
+
+  deleteAlert(key: string) {
+    this.ngProgress.start();
+    this.systemAlertService.deleteAlert(key).
+    subscribe(
+      (data: any) => {
+        this.ngProgress.done();
+        this.reenableButton.emit(false);
+
+        // Remove from local arrays of systemAlerts
+        this.systemAlerts = this.systemAlerts.filter(alert => alert.key !== key);  
+        this.alertService.success(data.message, true);
+      },
+      error => {
+        this.ngProgress.done();
+        this.reenableButton.emit(false);
+        if (error.status === 0) {
+          console.log('service down ', error);
+        } else {
+          this.alertService.error(error.statusText);
+        }
+      }
+    );
+  }
+
+  getButtonText(message: string) {
+    if (message.includes('restarted')) {
+      return "Show Logs";
+    }
+    if (message.includes('updates')) {
+      return "Upgrade";
```

Comment:
To make future proof, in next PR - return "" at the end of this function as default condition and if empty then don't show the corresponding action button